### PR TITLE
Manual: mention in hacking docs

### PR DESCRIPTION
**Pull request by grahamc**

### CODE REVIEW

File: doc/manual/src/hacking.md [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
     $ cd hydra
 
-To build it and its dependencies:
-
-    $ nix-build release.nix -A build.x86_64-linux
-
-To build all dependencies and start a shell in which all environment
-variables (such as PERL5LIB) are set up so that those dependencies can
-be found:
+To enter a shell in which all environment variables (such as PERL5LIB)
```
Comment (SuperSandro2000) [inline doc/manual/src/hacking.md:9]:
```suggestion
To enter a shell in which all environment variables (such as ``PERL5LIB``)
```

I tend to add code blocks to all paths and envs to make it easier to spot them. Maybe something to adapt.

File: foreman/start-manual.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+mdbook serve \
+  --port 63332 \
+  --dest-dir $(pwd)/.hydra-data/manual \
+  $(pwd)/doc/manual/
```
Comment (SuperSandro2000) [inline foreman/start-manual.sh:6]:
```suggestion
  --dest-dir .hydra-data/manual \
  doc/manual/
```

Is there any advantage of adding pwd explicitly?

File: foreman/start-manual.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
```
Comment (SuperSandro2000) [inline foreman/start-manual.sh:1]: Would it be a good idea to use nix-shell here?

File: doc/manual/src/hacking.md [hunk 4]
```diff
@@ -44,3 +43,15 @@ in the root of the repository at least once.
...
+
+By default, `foreman start` runs mdbook in "watch" mode. mdbook listens
+at [http://localhost:63332/](http://localhost:63332/) by default, and
+will reload the page every time you save.
+
+## Building
+
+To build Hydra and its dependencies:
+
+    $ nix-build release.nix -A build.x86_64-linux
```
Comment (SuperSandro2000) [inline doc/manual/src/hacking.md:57]: Are there other platforms a user might want to know about?

File: Procfile [hunk 5]
```diff
@@ -3,3 +3,4 @@ hydra-queue-runner: ./foreman/start-queue-runner.sh
 hydra-evaluator: ./foreman/start-evaluator.sh
 hydra-notify: ./foreman/start-notify.sh
 postgres: ./foreman/start-postgres.sh
+manual: ./foreman/start-manual.sh
```
Comment (SuperSandro2000) [inline Procfile:6]: Maybe this should be sorted alphabetically.

File: doc/manual/src/hacking.md [hunk 6]
```diff
@@ -23,7 +17,12 @@ To build Hydra, you should then do:
     [nix-shell]$ configurePhase
     [nix-shell]$ make
 
-You can run the Hydra web server in your source tree as follows:
+You start a local database, the webserver, and other components with
+foreman:
+
+    $ foreman start
+
```
Comment (SuperSandro2000) [inline doc/manual/src/hacking.md:24]:
```suggestion

```ShellSession
$ foreman start
```

```

That should improve highlighting on github and maybe mdbook.

File: foreman/start-manual.sh (hunk 3 above)
Comment (grahamc) [reply foreman/start-manual.sh:1]: I think it isn't needed: it'll slow down the startup time and foreman start isn't expected to work at all outside of a nix shell.

File: doc/manual/src/hacking.md (hunk 4 above)
Comment (grahamc) [reply doc/manual/src/hacking.md:57]: None that I know of where people actually deploy Hydra today...

File: doc/manual/src/hacking.md [hunk 7]
```diff
@@ -44,3 +43,15 @@ in the root of the repository at least once.
 if run with high parallelism [due to an issue in
 `Test::PostgreSQL`](https://github.com/TJC/Test-postgresql/issues/40)
 causing database ports to collide.
+
+## Working on the Manual
+
+By default, `foreman start` runs mdbook in "watch" mode. mdbook listens
+at [http://localhost:63332/](http://localhost:63332/) by default, and
```
Comment (sternenseemann) [inline doc/manual/src/hacking.md:50]: Style: duplication of “by default”

File: doc/manual/src/hacking.md [hunk 8]
```diff
@@ -1,19 +1,13 @@
...
 
-To build it and its dependencies:
-
-    $ nix-build release.nix -A build.x86_64-linux
-
-To build all dependencies and start a shell in which all environment
-variables (such as PERL5LIB) are set up so that those dependencies can
-be found:
+To enter a shell in which all environment variables (such as PERL5LIB)
+are set up so that those dependencies can be found:
```
Comment (sternenseemann) [inline doc/manual/src/hacking.md:10]: Reference: earlier mention of dependencies got removed.

File: doc/manual/src/hacking.md (hunk 1 above)
Comment (grahamc) [reply doc/manual/src/hacking.md]: Thanks, I usually do. This document was ported from XML and has some clean-up along these lines to be done :). Fixed!

File: foreman/start-hydra.sh [hunk 9]
```diff
@@ -5,6 +5,12 @@ while ! pg_isready -h $(pwd)/.hydra-data/postgres -p 64444; do sleep 1; done
 
 createdb -h $(pwd)/.hydra-data/postgres -p 64444 hydra
 
+# create a db for the default user. Not sure why, but
+# the terminal is otherwise spammed with:
+#
+#     FATAL:  database "USERNAME" does not exist
+createdb -h $(pwd)/.hydra-data/postgres -p 64444 "$(whoami)" || true
```
Comment (SuperSandro2000) [inline foreman/start-hydra.sh:12]: Not sure if these ``$(pwd)`` shoud also be converted.

File: doc/manual/src/projects.md [hunk 10]
```diff
@@ -230,11 +232,15 @@ yield an error like this:
...
 
-    $ nix-build -I ~/src release.nix -A tarball
+```console
+$ nix-build -I ~/src release.nix -A tarball
+```
 
 Similarly, the `build` jobset can be evaluated:
 
-    $ nix-build -I ~/src release.nix -A build
+```console
```
Comment (SuperSandro2000) [inline doc/manual/src/projects.md:241]: Not sure if there is any difference between ShellSession and console.

File: doc/manual/src/installation.md [hunk 11]
```diff
@@ -158,8 +172,10 @@ some additional configuration must be made.
 
 Edit your `hydra.conf` file in a similar way to this example:
 
-    using_frontend_proxy 1
-    base_uri example.com
+```
```
Comment (SuperSandro2000) [inline doc/manual/src/installation.md:175]:
```suggestion
```conf
```

File: foreman/start-hydra.sh (hunk 9 above)
Comment (grahamc) [reply foreman/start-hydra.sh:12]: Nah, I have a PR brewing that'll clean these up. Thanks, though.

File: doc/manual/src/projects.md (hunk 10 above)
Comment (grahamc) [reply doc/manual/src/projects.md:241]: mdbook doesn't support ShellSession, but does support console, and GitHub treats console as an alias to ShellSession. :+1:

**Pull request merged**